gokart/tree/task_info_formatter.py

```diff
@@ -36,7 +37,26 @@ def task_info_dict(self):
                     params=self.params,
                     processing_time=self.processing_time,
                     is_complete=self.is_complete,
-                    task_log=self.task_log)
+                    task_log=self.task_log,
+                    requires=self.requires)
+
+
+class RequiredTask(NamedTuple):
+    name: str
+    unique_id: str
+
+
+def _make_requires_info(requires):
+    if isinstance(requires, TaskOnKart):
+        return RequiredTask(name=requires.__class__.__name__, unique_id=requires.make_unique_id())
+
+    if isinstance(requires, list) or isinstance(requires, tuple):
+        return [_make_requires_info(requires=item) for item in requires]
+
+    if isinstance(requires, dict):
+        return {key: _make_requires_info(requires=item) for key, item in requires.items()}
+
+    raise TypeError(f'`requires` has unexpected type {type(requires)}. Must be `TaskOnKart`, `List[TaskOnKart]`, or `Dict[str, TaskOnKart]`')
```

Comment:
IMO, raising TypeError case test is needed

Reply:
I've added test_task_info_formatter.py for testing _make_requires_info() !

Reply:
Thanks! Confirmed